### ISSUE TITLE
False positive for selector-type-no-unknown

### ISSUE THREAD

**Issue by SDSLeon** [stylelint/stylelint#2336]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

For code like this rule selector-type-no-unknown show error:

> Which rule, if any, is this issue related to?

selector-type-no-unknown

> What CSS is needed to reproduce this issue?

```css
input {
        &:not(checked) {
            position: absolute;
            opacity: 0;
        }
}
```

> What stylelint configuration is needed to reproduce this issue?

stylelint-config-standard

> Which version of stylelint are you using?

e.g. `^7.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

webpack plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

e.g. "Yes, it's related to postCSS with nested"

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following error were flagged: Unexpected unknown type selector "checked"

**Comment by evilebottnawi** [comment]
@SDSLeon Thanks for feedback, accepted.

**Comment by hudochenkov** [comment]
Thank you for a report. Are you sure you wrote your CSS correctly? I believe correct way is with a colon before `checked` if you want to target non-checked inputs:

```scss
input {
        &:not(:checked) {
            position: absolute;
            opacity: 0;
        }
}
```

http://codepen.io/hudochenkov/pen/PWxbJd

**Comment by evilebottnawi** [comment]
@hudochenkov oh yep, it slipped from my mind :smile: , but be good added more test.

**Comment by evilebottnawi** [comment]
Rule works as expected. I am add more tests.
